### DISCUSSION TITLE
How do I update a record's datetime field via extend with Go?

### DISCUSSION THREAD

**Discussion by willbraun** [pocketbase/pocketbase#4101] [Q&A] [answered]
Hi all, I am trying to update a record's datetime value using the extend with Go option. I am trying to set the value of "prediction_close" from null to a time.Time value, but the updates aren't taking. Debug messages are logging so I know it's running and I've tried formatting newTime as a string as well, so I feel like I'm missing something very simple. Any ideas?

Here is the gist of my code.

```
	app.OnRecordAfterUpdateRequest("draw_slot").Add(func(e *core.RecordUpdateEvent) error {
		drawId := e.Record.GetString("draw_id")
		record, err := app.Dao().FindRecordById("draw", drawId)
		if err != nil {
			log.Panicln(err)
		}

		newTime := time.Now().Add(12 * time.Hour)
		record.Set("prediction_close", newTime)

		return nil
	})
```

**Comment by ganigeorgiev** [comment, marked as answer]
The `after*` hooks run **after** the model was persisted.
If you want to modify a record field in an after hook you need to persist it manually - `app.Dao().SaveRecord(record)`.

**Reply by ganigeorgiev** [reply]
Also you are fetching a different record from the one in `e.Record` so even if the above was part of a `before*` hook you still will have to resave it manually.

**Reply by willbraun** [reply]
@ganigeorgiev Thank you so much for the quick reply and explanation. This fixed my problem, I totally overlooked this in the documentation.
